runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -257,16 +259,19 @@ public int compareTo(CorfuQueueRecord<? extends E> o) {
         List<CorfuQueueRecord<E>> copy = new ArrayList<>(
                 Math.min(corfuTable.size(), maxEntries)
         );
+
         int index = 0;
-        for (Map.Entry<Long, E> entry : corfuTable.entrySet()) {
+        for (Long entryId : corfuTable.keySet().stream().sorted().collect(Collectors.toList())) {
```

Comment:
Should we filter keys out before collecting to a list?

Reply:
For maxEntries, maybe we can use Stream.limit()

Reply:
I thought about this too, but if we specify stream.limit() then it is possible that the stream has only a small number of entries and if the user has specified a start point towards the end, then it will miss out on the entries.
here is the example,
Q: 1, 2, 3, 4, 5
User wants 2 entries AFTER item number 3:
We should return: 4, 5
With Stream.limit() our collect() will have only 2 entries : 1, 2
which means 4, 5 cannot be processed.
Hope this helps?

Reply:
Fixed, I have used the stream.filter as suggested. thanks!